addons/mail/static/tests/web/chatter_topbar_tests.js

```diff
@@ -31,6 +31,66 @@ QUnit.test("base rendering", async (assert) => {
     assert.containsOnce($, ".o-mail-Followers");
 });
 
+QUnit.test("rendering when no access rights of chatter thread", async (assert) => {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({});
+    pyEnv["res.partner"].write([partnerId], { message_follower_ids: false });
+    const { openView } = await start({
+        async mockRPC(route, args, performRPC) {
+            const res = await performRPC(route, args);
+            if (route === "/mail/thread/data") {
+                res.hasWriteAccess = false;
+                res.hasReadAccess = false;
+            }
+            return res;
+        },
+    });
+    await openView({
+        res_id: partnerId,
+        res_model: "res.partner",
+        views: [[false, "form"]],
+    });
+    assert.containsOnce($, ".o-mail-Chatter-topbar");
+    assert.ok($(".o-mail-Chatter-sendMessage")[0].attributes.disabled);
+    assert.ok($(".o-mail-Chatter-logNote")[0].attributes.disabled);
+    assert.ok($(".o-mail-Chatter-activity")[0].attributes.disabled);
+    assert.ok($(".o-mail-Chatter-attachFiles")[0].attributes.disabled);
+});
+
+QUnit.test("attach files button rendering when no write access", async (assert) => {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({});
+    pyEnv["res.partner"].write([partnerId], { message_follower_ids: false });
+    pyEnv["ir.attachment"].create([
+        {
+            mimetype: "text/plain",
+            name: "Blah.txt",
+            res_id: partnerId,
+            res_model: "res.partner",
+        },
+    ]);
+    const { openView } = await start({
+        async mockRPC(route, args, performRPC) {
+            const res = await performRPC(route, args);
+            if (route === "/mail/thread/data") {
+                res.hasWriteAccess = false;
+                res.hasReadAccess = false;
+            }
+            return res;
+        },
+    });
+    await openView({
+        res_id: partnerId,
```

Comment:
The value of these tests are poor.
I would keep only a single test: _"Cannot post on thread with no access rights"_.
The code is shared between discuss, chat window, and chatter. Duplicating code adds more coverage, but it's not worth the maintenance cost of these tests.